tests/components/synology_dsm/test_media_source.py

```diff
@@ -0,0 +1,304 @@
+                return_value=[
+                    SynoPhotosItem(10, "", "filename.jpg", 12345, "10_1298753", "sm")
+                ]
+            ),
+            get_item_thumbnail_url=AsyncMock(return_value="http://my.thumbnail.url"),
+        )
+
+    return dsm
+
+
+@pytest.mark.usefixtures("setup_media_source")
+async def test_get_media_source(hass: HomeAssistant) -> None:
+    """Test the async_get_media_source function and SynologyPhotosMediaSource constructor."""
+
+    source = await async_get_media_source(hass)
+    assert isinstance(source, SynologyPhotosMediaSource)
+    assert source.domain == DOMAIN
+
+
+@pytest.mark.usefixtures("setup_media_source")
+@pytest.mark.parametrize(
+    ("identifier", "exception_msg"),
+    [
+        ("unique_id", "No album id"),
+        ("unique_id/1", "No file name"),
+        ("unique_id/1/cache_key", "No file name"),
+        ("unique_id/1/cache_key/filename", "No file extension"),
+    ],
+)
+async def test_resolve_media_bad_identifier(
+    hass: HomeAssistant, identifier: str, exception_msg: str
+) -> None:
+    """Test resolve_media with bad identifiers."""
+    source = await async_get_media_source(hass)
+    item = MediaSourceItem(hass, DOMAIN, identifier, None)
+    with pytest.raises(Unresolvable, match=exception_msg):
+        await source.async_resolve_media(item)
+
+
+@pytest.mark.usefixtures("setup_media_source")
+@pytest.mark.parametrize(
+    ("identifier", "url", "mime_type"),
+    [
+        (
+            "ABC012345/10/27643_876876/filename.jpg",
+            "/synology_dsm/ABC012345/27643_876876/filename.jpg",
+            "image/jpeg",
+        ),
+        (
+            "ABC012345/12/12631_47189/filename.png",
+            "/synology_dsm/ABC012345/12631_47189/filename.png",
+            "image/png",
+        ),
+    ],
+)
+async def test_resolve_media_success(
+    hass: HomeAssistant, identifier: str, url: str, mime_type: str
+) -> None:
+    """Test successful resolving an item."""
+    source = await async_get_media_source(hass)
+    item = MediaSourceItem(hass, DOMAIN, identifier, None)
+    result = await source.async_resolve_media(item)
+
+    assert result.url == url
+    assert result.mime_type == mime_type
+
+
+@pytest.mark.usefixtures("setup_media_source")
+async def test_browse_media_unconfigured(hass: HomeAssistant) -> None:
+    """Test browse_media without any devices being configured."""
+    source = await async_get_media_source(hass)
+    item = MediaSourceItem(
+        hass, DOMAIN, "unique_id/album_id/cache_key/filename.jpg", None
+    )
+    with pytest.raises(BrowseError, match="Diskstation not initialized"):
+        await source.async_browse_media(item)
+
+
+@pytest.mark.usefixtures("setup_media_source")
+async def test_browse_media_album_error(hass: HomeAssistant, dsm_with_photos) -> None:
+    """Test browse_media with unknown album."""
+    dsm = SynologyDSMData
+    dsm.api = dsm_with_photos
+    hass.data[DOMAIN] = {"unique_id": dsm}
```

Comment:
You could borrow some of the code from the options flow in `test_config_flow.py` and patch out the library just enough to get it to setup

Reply:
Can you maybe point me in the correct direction? I'm not entirely familiar with the tests. 

What is a better way on passing the data objects in the tests?

Just creating a setData function, and in the case it is called during testing, it sets the data with the test data objects, otherwise use the hass.data[DOMAIN]?

Reply:
@mib1185 Hey, I tried looking at the tests, but am a bit stuck, can you help a bit? Sorry

Reply:
i'll have a look at the weekend